src/components/table/use_table_scroll.ts

```diff
@@ -0,0 +1,53 @@
+import type { RowData, Table } from '@tanstack/react-table';
+import type {
+  ScrollToOptions as VirtualScrollToOptions,
+  Virtualizer,
+} from '@tanstack/react-virtual';
+import { useEffect, useRef } from 'react';
+
+import type { TableProps } from './table_root.js';
+
+export function useTableScroll<TData extends RowData>(
+  tableProps: TableProps<TData>,
+  table: Table<TData>,
+  virtualizer: Virtualizer<HTMLDivElement, Element>,
+) {
+  const tableRef = useRef<HTMLTableElement>(null);
+  const { scrollToRowRef, virtualizeRows } = tableProps;
+  useEffect(() => {
+    if (scrollToRowRef) {
+      if (virtualizeRows) {
+        scrollToRowRef.current = (
```

Comment:
Are you sure this is right?
So far I've always seen and used [`useImperativeHandle`](https://react.dev/reference/react/useImperativeHandle) to manipulate refs.
The official docs even mention scrolling to a node as a use case.

Reply:
Yes you are right! This is how refs should be exposed to their parent.